lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -15,6 +17,49 @@
 Applies DSC Resources by generating a configuration file and applying it.
 EOT
 
+  def initialize(value={})
+    super(value)
+    set_symlink_folder
+  end
+
+  if Puppet::Util::Platform.windows?
```

Comment:
Darn, looks like the methodology for Win/Linux in colors.rb was similar - https://github.com/puppetlabs/puppet/blob/3.8.0/lib/puppet/util/colors.rb#L79-L190